File: src/main/java/it/carloni/luca/aurora/utils/DateFormat.java
package it.carloni.luca.aurora.utils;

import java.time.format.DateTimeFormatter;

public enum DateFormat {

    DT_BUSINESS_DATE("yyyy-MM-dd", DateTimeFormatter.ISO_LOCAL_DATE);

    private final String format;
    private final DateTimeFormatter formatter;

    public DateTimeFormatter getFormatter() { return formatter; }
    public String getFormat() { return format; }

    DateFormat(String format, DateTimeFormatter dateTimeFormatter) {

        this.format = format;
        this.formatter = dateTimeFormatter;
    }
}

File: src/main/scala/it/luca/aurora/utils/DateFormat.scala
package it.luca.aurora.utils

import java.time.format.DateTimeFormatter

object DateFormat extends Enumeration {

  protected case class Val(format: String, formatter: DateTimeFormatter)
    extends super.Val

  import scala.language.implicitConversions
  implicit def valueToDateFormatVal(x: Value): Val = x.asInstanceOf[Val]

  val DtRiferimento: Val = Val("yyyy-MM-dd", DateTimeFormatter.ISO_LOCAL_DATE)

}
